Support attaching multiple policies via the parameter

zsims opened the pull request:
This closes #442 by supporting multiple (comma separated) ARNs in the existing `ManagedPolicyARN` parameter.

# Test with no ECR access flagged:

```bash
$ aws cloudformation create-stack --stack-name multiple-arns --template-body file://build/aws-stack.json --parameters "ParameterKey=BuildkiteAgentToken,ParameterValue=secret" "ParameterKey=ManagedPolicyARN,ParameterValue=arn:aws:iam::1234:policy/zacs-agents-AccessBuildAgencyResourcesPolicy-XXX,arn:aws:iam::1234:policy/serverless-gql-policy" --capabilities CAPABILITY_NAMED_IAM
$ aws iam list-attached-role-policies --role-name multiple-arns-Role
{
    "AttachedPolicies": [
        {
            "PolicyName": "zacs-agents-AccessBuildAgencyResourcesPolicy-XXX",
            "PolicyArn": "arn:aws:iam::1234:policy/zacs-agents-AccessBuildAgencyResourcesPolicy-XXX"
        },
        {
            "PolicyName": "serverless-gql-policy",
            "PolicyArn": "arn:aws:iam::1234:policy/serverless-gql-policy"
        }
    ]
}
```

# Test with custom managed policies + ECR access flagged
```bash
$ aws cloudformation create-stack --stack-name multiple-arns-with-ecr --template-body file://build/aws-stack.json --parameters "ParameterKey=BuildkiteAgentToken,ParameterValue=secret" "ParameterKey=ECRAccessPolicy,ParameterValue=poweruser" "ParameterKey=ManagedPolicyARN,ParameterValue=arn:aws:iam::1234:policy/zacs-agents-AccessBuildAgencyResourcesPolicy-XXX,arn:aws:iam::1234:policy/serverless-gql-policy" --capabilities CAPABILITY_NAMED_IAM
$ aws iam list-attached-role-policies --role-name multiple-arns-with-ecr-Role
{
    "AttachedPolicies": [
        {
            "PolicyName": "zacs-agents-AccessBuildAgencyResourcesPolicy-XXX",
            "PolicyArn": "arn:aws:iam::1234:policy/zacs-agents-AccessBuildAgencyResourcesPolicy-XXX"
        },
        {
            "PolicyName": "serverless-gql-policy",
            "PolicyArn": "arn:aws:iam::1234:policy/serverless-gql-policy"
        },
        {
            "PolicyName": "AmazonEC2ContainerRegistryPowerUser",
            "PolicyArn": "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryPowerUser"
        }
    ]
}
```

lox commented on templates/buildkite-elastic.yml:
```diff
@@ -367,15 +367,21 @@ Resources:
+              - UseECR
+              - { "Fn::FindInMap": [ ECRManagedPolicy, { Ref: ECRAccessPolicy }, 'Policy' ] }
+              - { Ref: 'AWS::NoValue' }
+            # This may support mutliple values of its own (separated by commas)
```
Typo in comment "mutliple"

zsims replied:
Oops. Fixed!

zsims commented:
I've also rebased this as the `keyName` PR landed before this

lox commented:
I wonder if we could make the type a `CommaDelimitedList` @zsims? My only concern with this approach is that we start to go down an approach where we aren't using typed CloudFormation inputs, which means input validation starts to be a thing.

zsims commented:
That's a good idea, should solve both questions. I'll test locally and report back.

zsims commented:
So further testing showed that this approach doesn't support attaching *no* managed policies, e.g. as [!Split](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-split.html) seems to return an empty string for `!Split [',' '']` rather than `AWS::NoValue` (which makes sense).

The CommaDelimitedList works great, just will need to think about the 0-state

zsims commented on templates/buildkite-elastic.yml:
```diff
@@ -378,15 +378,21 @@ Resources:
-          - { "Fn::FindInMap": [ ECRManagedPolicy, { Ref: ECRAccessPolicy }, 'Policy' ] }
-          - { Ref: 'AWS::NoValue' }
+      # Support multiple policies to attach by merging the values together and splitting on ','
+      ManagedPolicyArns: !Split
```
Per the PR discussion, this produces an empty string (thus not an ARN) when no policies are to be attached -- thus needs more thought :(

zsims commented:
Updated to handle the "no policy scenario", tested **successfully** in all these scenarios:

 1. No policies => Success
 2. Just the ecr policy (`ECRAccessPolicy` = `readonly`) => Success
 3. ECR policy + managed policy (`ECRAccessPolicy` = `readonly`, `ManagedPolicyARN` = `arn:aws:iam::aws:policy/AmazonAthenaFullAccess`) => Success
 4.  Single managed policy (`ManagedPolicyARN` = `arn:aws:iam::aws:policy/AmazonAthenaFullAccess`) => Success
 5. Multiple managed policies (`ManagedPolicyARN` = `arn:aws:iam::aws:policy/AmazonAthenaFullAccess,arn:aws:iam::aws:policy/AmazonChimeFullAccess`) => Success
 6. ECR policy + multiple managed policies (`ECRAccessPolicy` = `readonly`, `ManagedPolicyARN` = `arn:aws:iam::aws:policy/AmazonAthenaFullAccess,arn:aws:iam::aws:policy/AmazonChimeFullAccess`)
 7. Managed policies with spaces (`ManagedPolicyARN` = `arn:aws:iam::aws:policy/AmazonAthenaFullAccess,     arn:aws:iam::aws:policy/AmazonChimeFullAccess`) => Success (these all get stripped)

Ready for re-review @lox @toolmantim
